### PR TITLE
fix(branding): Fix dynamic logo display

### PR DESCRIPTION
**Pull request by vp8x8**
* Display of the logo has been reworked (simplified).
* The logo will not be displayed if the call to `branding` endpoint fails (for vpaas meetings).
* Add more docs.

### CODE REVIEW

File: react/features/base/react/components/web/Watermarks.js [hunk 1]
```diff
@@ -319,27 +221,52 @@ class Watermarks extends Component<Props, State> {
...
-    const { room } = state['features/base/conference'];
+    const {
+        customizationReady,
+        customizationFailed,
+        defaultBranding,
+        useDynamicBrandingData,
+        logoClickUrl,
+        logoImageUrl
+    } = state['features/dynamic-branding'];
+    const welcomePageVisible = !state['features/base/conference'].room;
```
Comment (saghul) [inline react/features/base/react/components/web/Watermarks.js:237]: Use isValidRoom please.

File: react/features/base/react/components/web/Watermarks.js [hunk 2]
```diff
@@ -319,27 +221,52 @@ class Watermarks extends Component<Props, State> {
...
+    } = interfaceConfig;
+    let _showJitsiWatermark = (!filmStripOnly
+          && (customizationReady && !customizationFailed)
+          && (SHOW_JITSI_WATERMARK || (isGuest && SHOW_JITSI_WATERMARK_FOR_GUESTS)))
+    || welcomePageVisible;
+    let _logoUrl = logoImageUrl;
+    let _logoLink = logoClickUrl;
+
+    if (useDynamicBrandingData) {
+        if (isVpaasMeeting(state)) {
```
Comment (saghul) [inline react/features/base/react/components/web/Watermarks.js:253]: Does this mean that people using their own deployments and relaying on dynamic branding will no longer be able to use it this way? We know people are already using this, dynamic branding should apply always IMHO.

**Review by saghul — COMMENTED** [review]
Left some comments.

File: react/features/base/react/components/web/Watermarks.js (hunk 2 above)
Comment (vp8x8) [reply react/features/base/react/components/web/Watermarks.js:253]: No. Everything should work as before. 

For `vpaas` meetings though, we don't show the logo if it was not explicitly set or the request has failed (these were _specs_ I received).

Comment (saghul) [reply react/features/base/react/components/web/Watermarks.js:253]: Ack

**Review by saghul — APPROVED** [review]

**Pull request merged**